File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    dict = {}
    answer = 1
    for name, type in clothes:
        if type in dict:
            dict[type].append(name)
        else:
            dict[type] = [name]
    
    for d in dict:
        answer *= (len(dict[d]) + 1) 
    answer -= 1
    
    return answer

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    var answer = 1;
    let map = new Map();
    clothes.forEach(([name, type]) => {
        if(map.has(type)) {
            map.set(type, map.get(type) + 1);
        } else map.set(type, 1);
    })
    
    map.forEach(value => {
        answer *= value + 1
    })
    
    return answer - 1;
}